account_payment_order/models/account_payment_order.py

```diff
@@ -162,10 +146,18 @@ def create(self, vals):
     @api.onchange('payment_mode_id')
     def payment_mode_id_change(self):
         journal_id = False
+        res = {'domain': {
+            'journal_id': "[('id', '=', False)]",
```

Comment:
Then please hide the field when no payment mode is selected in the view via `attrs` attribute.